qbraid_qir/qasm3/visitor.py

```diff
@@ -1149,6 +1161,100 @@ def _visit_statement_block(block):
             one=lambda: _visit_statement_block(if_block),
         )
 
+    def _extract_values_from_discrete_set(self, discrete_set: DiscreteSet) -> List[int]:
+        """Extract the values from a discrete set.
+
+        Args:
+            discrete_set (DiscreteSet): The discrete set to extract values from.
+
+        Returns:
+            List[int]: The extracted values.
+        """
+        values = []
+        for value in discrete_set.values:
+            if not isinstance(value, IntegerLiteral):
+                self._print_err_location(discrete_set.span)
+                raise Qasm3ConversionError(
+                    f"Unsupported discrete set value {value} in discrete set"
+                )
+            values.append(value.value)
+        return values
+
+    def _visit_alias_statement(self, statement: AliasStatement) -> None:
+        """Visit an alias statement element.
+
+        Args:
+            statement (AliasStatement): The alias statement to visit.
+
+        Returns:
+            None
+        """
+        # pylint: disable=too-many-branches
+        target = statement.target
+        value = statement.value
+
+        alias_reg_name = target.name
+        alias_reg_size = None
+        aliased_reg_name = None
+        aliased_reg_size = None
+
+        if isinstance(value, Identifier):
+            aliased_reg_name = value.name
+        elif isinstance(value, IndexExpression):
+            aliased_reg_name = value.collection.name
+        else:
+            self._print_err_location(statement.span)
+            raise Qasm3ConversionError(
+                f"Unsupported aliasing {statement} not supported at the moment"
+            )
+
+        if aliased_reg_name not in self._qreg_size_map:
+            self._print_err_location(statement.span)
+            raise Qasm3ConversionError(f"Qubit register {aliased_reg_name} not found for aliasing")
+
+        aliased_reg_size = self._qreg_size_map[aliased_reg_name]
+        if isinstance(value, Identifier):  # "let alias = q;"
+            for i in range(aliased_reg_size):
+                self._qubit_labels[f"{alias_reg_name}_{i}"] = self._qubit_labels[
+                    f"{aliased_reg_name}_{i}"
+                ]
+            alias_reg_size = aliased_reg_size
+        elif isinstance(value, IndexExpression):
+            if isinstance(value.index, DiscreteSet):  # "let alias = q[{0,1}];"
+                qids = self._extract_values_from_discrete_set(value.index)
```

Comment:
I think we should have bounds check here, as the `DiscreteSet` might contain invalid indices. Although the dict containing qubit labels would raise a `KeyError`, it would be nice to throw a verbose error message